pkg/gadgets/trace/tcpdrop/tracer/bpf/tcpdrop.bpf.c

```diff
@@ -140,9 +141,14 @@ int ig_tcpdrop(struct trace_event_raw_kfree_skb *ctx)
 	struct sock *sk = BPF_CORE_READ(skb, sk);
 	int reason = ctx->reason;
 
-	// If bpf_core_enum_value fails, it will return 0 and that will not be a silent failure
+	// If enum value was not found, bpf_core_enum_value returns 0.
+	// The verifier will reject the program with
+	// invalid func unknown#195896080
+	// 195896080 == 0xbad2310 reads "bad relo"
 	int reason_not_specified = bpf_core_enum_value(
 		enum skb_drop_reason, SKB_DROP_REASON_NOT_SPECIFIED);
+	if (reason_not_specified == 0)
+		bpf_core_unreachable();
```

Comment:
> With which kernel the SKB_DROP_REASON_NOT_SPECIFIED was added?

https://github.com/torvalds/linux/commit/c504e5c2f9648a1e5c2be01e8c3f59d394192bd3 (Linux 5.17) introduces the enum:
```patch
+enum skb_drop_reason {
+       SKB_DROP_REASON_NOT_SPECIFIED,
+       SKB_DROP_REASON_MAX,
+};
```

Then, https://github.com/torvalds/linux/commit/1330b6ef3313fcec577d2b020c290dc8b9f11f1a (Linux 5.18) adds `Adds SKB_NOT_DROPPED_YET`, so the value `SKB_DROP_REASON_NOT_SPECIFIED` shifts to 1:
```patch
 enum skb_drop_reason {
+       SKB_NOT_DROPPED_YET = 0,
        SKB_DROP_REASON_NOT_SPECIFIED,  /* drop reason is not specified */
```

I remember that the tracepoint was called often even without dropping packets, with `SKB_NOT_DROPPED_YET`. So I want to avoid that. This is similar to the bcc tool:
https://github.com/iovisor/bcc/blob/master/tools/tcpdrop.py#L182


> Because, with this new code, it basically means we will not load this eBPF program on kernel without this value.

True. That makes the requirement to Linux 5.18 instead of Linux 5.17. If we keep this, I should update docs/getting-started/requirements.md

> Is this really what we want?

Good question. I thought it would be better than risking invalid results on old kernels. What do you think?

Reply:
> It seems fine to me, as we are just dropping one kernel version, but yes definitely the documentation should be updated.

Ok, updated.

> I am also wondering if we cannot think to a function which would generate the bad relocation but also print a message indicate we did this on purpose because the eBPF code is not compatible with this kernel?

That would be good. Not sure how to do it in a  clean way. I could parse the verifier error message and return the comment in this way, but that's a hack:
https://gist.github.com/alban/b1bd4ec2fbaa7138022fd9f0482bf95e